src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -83,14 +89,54 @@ public OcrValidator(
                         authTokenGenerator.generate()
                     ))
                     .onSuccess(res -> handleValidationResponse(res, envelope, docWithOcr))
-                    .onFailure(exc -> handleRestClientException(exc, validationUrl, envelope, docWithOcr))
+                    .onFailure(exc -> handleRestClientException(
+                        exc,
+                        validationUrl,
+                        envelope,
+                        docWithOcr
+                    ))
                     .map(response -> ocrValidationWarnings(docWithOcr, response.warnings))
-                    .getOrElseGet(exc ->
-                        ocrValidationWarnings(
-                            docWithOcr,
-                            singletonList("OCR validation was not performed due to errors")
-                        )
+                    .getOrElseGet(exc -> handleOcrValidationError(
+                        exc,
+                        blobClient,
+                        leaseId,
+                        docWithOcr,
+                        validationUrl,
+                        envelope.zipFileName
+                    ))
+        );
+    }
+
+    private OcrValidationWarnings handleOcrValidationError(
+        Throwable ex,
+        BlobClient blobClient,
+        String leaseId,
+        InputScannableItem docWithOcr,
+        String validationUrl,
+        String zipFileName
+    ) {
+        if (ex instanceof HttpServerErrorException) {
+            if (ocrValidationRetryManager.setRetryDelayIfPossible(blobClient, leaseId)) {
+                throw new OcrValidationServerSideException(
+                    String.format(
+                        "Error calling validation endpoint. "
+                            + "Url: %s, DCN: %s, doc type: %s, doc subtype: %s, envelope: %s. "
+                            + "Error message: %s, status code: %s, status text: %s",
+                        validationUrl,
+                        docWithOcr.documentControlNumber,
+                        docWithOcr.documentType,
+                        docWithOcr.documentSubtype,
+                        zipFileName,
+                        ex.getMessage(),
+                        ((HttpServerErrorException) ex).getStatusCode().toString(),
+                        ((HttpServerErrorException) ex).getStatusText()
```

Comment:
Instead of extracting data from ex, couldn't the ex be passed as param to the exception? Will be much easier to trace as the proper stacktraces will be preserved (and other information).